src/xwayland.c

```diff
@@ -109,24 +109,16 @@ handle_commit(struct wl_listener *listener, void *data)
 	/* Must receive commit signal before accessing surface->current* */
 	struct wlr_surface_state *state = &view->surface->current;
 	struct wlr_box *current = &view->current;
-	struct wlr_box *pending = &view->pending;
 
-	if (current->width == state->width && current->height == state->height) {
-		return;
-	}
-
-	current->width = state->width;
-	current->height = state->height;
-
-	if (current->x != pending->x) {
-		/* Adjust x for queued up configure events */
-		current->x = pending->x + pending->width - current->width;
-	}
-	if (current->y != pending->y) {
-		/* Adjust y for queued up configure events */
-		current->y = pending->y + pending->height - current->height;
+	/*
+	 * If there is a pending move/resize, wait until the surface
+	 * size changes to update geometry. The hope is to update both
+	 * the position and the size of the view at the same time,
+	 * reducing visual glitches.
```

Comment:
> FIXME: please add a comment to explain what's going on with the configure serials here

It is basically doing the same thing as this comment explains for xwayland. The difference is that the wayland protocol with its configure serial is deterministic while the X11 variant requires guesswork based on size / position.

As far as I understand `view->pending_configure_serial` can have 3 states:
- `0`: no configure request pending
- `< xdg_surface->current.configure_serial`: some configure request still pending
- `== xdg_surface->current.configure_serial`: some configure request was pending but we now catched up and thus reset `view->pending_configure_serial` to `0`

Feel free to correct me when I got something wrong.

Reply:
I removed the FIXME; I think I understand how the configure serials work now. Thanks!